### PR TITLE
implement SUB opcode

### LINKED ISSUE #6: Opcode: SUB

### PR DESCRIPTION
**Pull request by FrancoGiachetta**
Closes #6 
This PR handles the implementation of SUB opecode

### CODE REVIEW

File: tests/operations.rs [hunk 1]
```diff
@@ -75,6 +75,18 @@ fn push_push_add() {
...
+#[test]
+fn push_push_sub() {
+    let (a, b) = (11, 31);
+
+    let program = vec![
+        Operation::Push32(new_32_byte_immediate(a)),
+        Operation::Push32(new_32_byte_immediate(b)),
+        Operation::Sub,
+    ];
+    run_program_assert_result(program, b - a);
```
Comment (juanbono) [inline tests/operations.rs:87]: check if the subtraction wraps the result when you do : `0 - 10` or something like that. Please add a test like this one for this case.

Comment (juanbono) [reply tests/operations.rs:87]: same for U256::MAX + 10

**Review by juanbono — CHANGES_REQUESTED** [review]
see comments

File: tests/operations.rs (hunk 1 above)
Comment (FrancoGiachetta) [reply tests/operations.rs:87]: Checking

Comment (FrancoGiachetta) [reply tests/operations.rs:87]: How could I represent U256::MAX?

Comment (FrancoGiachetta) [reply tests/operations.rs:87]: Screenshot 2024-05-29 at 5 08 34 PM
It's giving me this

Comment (MegaRedHand) [reply tests/operations.rs:88]: You can build the array: `[0xFF; 32]`, and calculate the result manually.

File: tests/operations.rs [hunk 2]
```diff
@@ -75,6 +76,48 @@ fn push_push_add() {
...
+        Operation::Push32(new_32_byte_immediate(a)),
+        Operation::Push32(new_32_byte_immediate(b)),
+        Operation::Sub,
+    ];
+
+    let result = ((b as u32).wrapping_sub(a as u32)) as u8;
+
+    run_program_assert_result(program, result);
+}
+
```
Comment (MegaRedHand) [inline tests/operations.rs:105]: You're missing the `#[test]` tag

Comment (FrancoGiachetta) [reply tests/operations.rs:105]: My bad, fixing it

Comment (FrancoGiachetta) [reply tests/operations.rs:105]: Fixed

**Review by juanbono — APPROVED** [review]

**Pull request merged**